src/large_pages/node_large_page.cc

```diff
@@ -282,20 +294,42 @@ bool IsSuperPagesEnabled() {
 }
 #endif
 
+// Functions in this class must always be inlined because they must end up in
+// the `lpstub` section rather than the `.text` section.
+class MemoryMapPointer {
+ public:
+  FORCE_INLINE MemoryMapPointer() {}
+  FORCE_INLINE MemoryMapPointer(void* start,
+                                size_t size,
+                                int prot,
+                                int flags,
+                                int fd = -1,
+                                int offset = 0) {
+    mem_ = mmap(start, size, prot, flags, fd, offset);
+    size_ = size;
+  }
+  FORCE_INLINE bool operator==(void* rhs) const { return mem_ == rhs; }
+  FORCE_INLINE operator void*() const { return mem_; }
+  FORCE_INLINE void operator= (MemoryMapPointer&& other) {
+    this->~MemoryMapPointer();
+    mem_ = other.mem_;
+    size_ = other.size_;
+    other.mem_ = nullptr;
+    other.size_ = 0;
+  }
+  FORCE_INLINE void operator= (void* rhs) { mem_ = rhs; }
+  FORCE_INLINE ~MemoryMapPointer() {
+    if (mem_ != nullptr && mem_ != MAP_FAILED && munmap(mem_, size_) != 0)
+      PrintSystemError(errno);
```

Comment:
This is just stylistic but I find it a little easier to read with early-return guards:
```suggestion
    if (mem_ == nullptr) return;
    if (mem_ == MAP_FAILED) return;
    if (munmap(mem_, size_) == 0) return;
    PrintSystemError(errno);
```

Reply:
Done.